c7n/resources/asg.py

```diff
@@ -157,13 +157,13 @@ class LaunchConfigFilter(ValueFilter, LaunchConfigFilterBase):
 
     .. code-block:: yaml
 
-            policies:
-              - name: launch-config-public-ip
-                resource: asg
-                filters:
-                  - type: launch-config
-                    key: AssociatePublicIpAddress
-                    value: true
+        policies:
+          - name: asg-launch-config-using-image
+            resource: asg
+            filters:
+              - type: launch-config
+                key: ImageId
```

Comment:
The non public up is probably a more useful example

Reply:
i changed this one as i was unable to get the `AssociatePublicIpAddress` to actually give me a valid return. when i look at what the `LaunchConfigFilterBase` pulls out, i do not see that value in the response.

Reply:
AssociatePublicIpAddress is an attribute of the launch config per aws docs.

Reply:
bah, yes, ok, so it is a valid `key` if it is specified in the launch-config... if you don't select `Assign a public IP address to every instance.` that key doesn't seem to be attached... so it's either `true` or `absent` for a valid result.